test/Generics/requirement_machine_diagnostics.swift

```diff
@@ -197,4 +197,112 @@ func inferred6<T : P11>(_: T) where T.Y : Hashable, T.Z == Set<T.X>, T.X == T.Y
 
 func typeMatcherSugar<T>(_: T) where Array<Int> == Array<T>, Array<Int> == Array<T> {}
 // expected-warning@-1 2{{redundant same-type constraint 'Array<Int>' == 'Array<T>'}}
-// expected-warning@-2{{redundant same-type constraint 'T' == 'Int'}}
\ No newline at end of file
+// expected-warning@-2{{redundant same-type constraint 'T' == 'Int'}}
+
+// MARK: - Conflict diagnostics
+
+protocol ProtoAlias1 {
+  typealias A1 = Int
+}
+
+protocol ProtoAlias2 {
+  typealias A2 = String
+}
+
+func basicConflict<T: ProtoAlias1 & ProtoAlias2>(_:T) where T.A1 == T.A2 {}
+// expected-error@-1{{no type for 'T.A1' can satisfy both 'T.A1 == Int' and 'T.A1 == String'}}
```

Comment:
Oh, "no type for T" is a great way to say "type parameter T" without saying "type parameter". Nice!